Added Scripting guide by Bizzo in Markdown format

Lord-Baldwin-of-Kerry opened the pull request:
@Snailedlt Let me know if something needs to change

Snailedlt commented on docs/create-handlers.md:
````diff
@@ -3,14 +3,326 @@ id: create-handlers
+- If by doing manually the second instance starts, while through Nucleus-Coop don't, try adding this line. (You can also disable all the not necessary lines and enable one at the time between each test to identify which one causes problems). 
+```Game.HookInit = true;					//Enable or disable hooks of functions some games may try and use to prevent multiple instances from running | default: false```
+
+### Connection between instances
````
Don't change the title for now. First commit we should keep as close to the original as possible. then after that we can start changing it up :)
Change to H2 here: `## Connectivity`

Snailedlt commented on docs/create-handlers.md:
```diff
@@ -3,14 +3,326 @@ id: create-handlers
+
+
+
+### Scripting Tips
```
change to H2 `## Scripting Tips`

Snailedlt commented on docs/create-handlers.md:
```diff
@@ -3,14 +3,326 @@ id: create-handlers
+
+If all work is time to script it. Now, You will start testing input restriction and focus methods!
+Start by checking how the public scripts work and use as an example one that uses the same game-engine of the game you are trying to script.
+##### Some tips and essential lines
```
Change to H3 `### Some tips and essential lines`

Snailedlt commented on docs/create-handlers.md:
```diff
@@ -3,14 +3,326 @@ id: create-handlers
+
+Here some tips and essential lines:
+
+#### Game Info
```
Change to H3 `### Game Info`

Snailedlt commented on docs/create-handlers.md:
```diff
@@ -3,14 +3,326 @@ id: create-handlers
+Here some tips and essential lines:
+
+#### Game Info
+***These lines must be inside the script:***
```
Change to H4, and like I mentioned earlier, the name should stay the same for now `#### Essential lines`

Snailedlt commented on docs/create-handlers.md:
```diff
@@ -3,14 +3,326 @@ id: create-handlers
+Game.MaxPlayersOneMonitor = 4; |	This is just an info. It will not limit the players' numbers.
+Game.MaxPlayers = 16; | 	This is just an info. Usually, we write the max players number the game support. 
+
+***Other important ones:***
```
Change to H4 `#### Other important one`

Snailedlt commented on docs/create-handlers.md:
```diff
@@ -3,14 +3,326 @@ id: create-handlers
+Game.LauncherExe = "gamelauncher.exe";  | If the game needs to go through a launcher before opening. Nucleus will use this exe to start the game and it will follow Game.ExecutableName process for resizing and positioning. You can also write it like this: Game.LauncherExe = "gamelauncher.exe|NucleusDefined"; for let the User select the launcher exe the very first time (useful if is not located in  ...
+Game.LauncherExeIgnoreFileCheck = true;	 | Do not check if Launcher Exe exists in game folder | you will need to provide a relative filepath from game root folder
+
+***Files copies lines***
```
These look like a nice addition, but for now. Remove them, and add them in a new commit after instead

## Remove from here

Snailedlt commented on docs/create-handlers.md:
```diff
@@ -3,14 +3,326 @@ id: create-handlers
+Line | Description
+---- | -----------
+Game.KeepSymLinkOnExit = true; | False = after Nucleus is closed, game files inside Nucleus\content will be deleted. Enable = Nucleus will keep the files inside Nucleus\content even after is closed. Useful if the game store saves it in the game-folder.
+
```
## To here

Snailedlt commented on docs/create-handlers.md:
```diff
@@ -3,14 +3,326 @@ id: create-handlers
+---- | -----------
+Game.KeepSymLinkOnExit = true; | False = after Nucleus is closed, game files inside Nucleus\content will be deleted. Enable = Nucleus will keep the files inside Nucleus\content even after is closed. Useful if the game store saves it in the game-folder.
+
+## Nucleus-Coop Environment
```
Change to H4 `#### Nucleus-Coop Environment`

Snailedlt commented on docs/create-handlers.md:
```diff
@@ -3,14 +3,326 @@ id: create-handlers
+
+### Handlers
+
+***This is needed:***
```
Change to H4 `#### This is needed`

Snailedlt commented on docs/create-handlers.md:
```diff
@@ -3,14 +3,326 @@ id: create-handlers
+---- | ----------
+Game.HandlerInterval = 100; | The interval in milliseconds the Handler should be updated at. Set it to 0 to disable updating (will lose all functionality that depends on ticks).
+
+***Higher time could help low spec PCs:***
```
I like the title change here, so go ahead and keep it. Still needs to change to H4 heading though :) `#### Higher time could help low spec PCs`

Snailedlt commented on docs/create-handlers.md:
```diff
@@ -3,14 +3,326 @@ id: create-handlers
+---- | ----------
+Game.PauseBetweenStarts = 20; | Pause between game intances starts in milliseconds.
+
+### Focus
```
Good! :D

Snailedlt commented on docs/create-handlers.md:
```diff
@@ -3,14 +3,326 @@ id: create-handlers
+Game.DocumentsSavePath = "Path\\Here"; | Relative path from user document folder (e.g. C:\Users\ZeroFox\Documents) to game's save path. \| Used to provide some extra functionality (open/delete/copy over to Nucleus Environment). Use this when the game uses Documents to store game files. In context = Context.DocumentsSavePath
+
+
+### Handlers
```
👌

Snailedlt commented on docs/create-handlers.md:
```diff
@@ -3,14 +3,326 @@ id: create-handlers
+---- | ----------
+Game.HookInit = true; | Enable or disable hooks of functions some games may try and use to prevent multiple instances from running \| default: false
+
+### Input
```
👌

Snailedlt commented on docs/create-handlers.md:
```diff
@@ -3,14 +3,326 @@ id: create-handlers
+
+#### Create a Directory
+
+Code | Steps
```
Very nice use of table headers :+1:

Snailedlt commented on docs/create-handlers.md:
```diff
@@ -3,14 +3,326 @@ id: create-handlers
+
+Line | Description
+---- | -----------
+Game.UseNucleusEnvironment = true; | Use custom environment variables for games that use them, replaces some common paths (e.g. AppData) with C:\Users\<your username>\NucleusCoop
```
This line causes an error because `<your username>` isn't escaped:
![image](https://user-images.githubusercontent.com/43886029/128763184-81455193-0520-496d-9978-6d432dc514db.png)

You can fix it by escaping the `<` and `>` signs:  `Game.UseNucleusEnvironment = true; | Use custom environment variables for games that use them, replaces some common paths (e.g. AppData) with C:\Users\\<your username\>\NucleusCoop`

Lord-Baldwin-of-Kerry commented:
I will start to make these changes!

Lord-Baldwin-of-Kerry commented:
@Snailedlt I added the images and the docusaurus flags( I can't seem to make then pop in my machine) lmk if anything else needs fixing! :D

Snailedlt commented:
@Lord-Baldwin-of-Kerry Great, I see a couple of things, which I will comment on shortly :) I see the images and the note tag here btw. Not sure what you meant with `docusaurus flags` though

Snailedlt commented on docs/create-handlers.md:
```diff
@@ -2,15 +2,330 @@
+Game.SupportsKeyboard = true; | If enabled, will allow the keyboard to be used as a device input in Nucleus. Before enabling it by default, be sure that the keyboard/mouse works without problems.
+Game.KeyboardPlayerFirst = false; | Keyboard player will be the first to be initialized
+
+Gamepads:
```
change to: 
`#### Gamepads`

This makes it easier to reference/link to, for support questions

Snailedlt commented on docs/create-handlers.md:
```diff
@@ -2,15 +2,330 @@
+Open Process Explorer (with ***admin rights***), start the game, click the game process on the list, and press ***View Handles*** (Ctrl+H). You will see the full list of mutex/event/semaphore, etc.
+If you think to have found the right one, right-click on it and choose ***"Close Handle"***. After that, try to run a second instance of the game and see if it opens. (some games require multiple mutexes to be closed).
+Be careful that launchers check if the game is already running. For example, Goldberg, make it unseen from Steam.
+#### Check public scripts for examples.
```
remove this line :)

Snailedlt commented on docs/create-handlers.md:
```diff
@@ -2,15 +2,330 @@
+#### Rename the game exe:
+Change the game exe name for each instance (in Nucleus-Coop = Game.ChangeExe = true;)
+![Rename game exe](../static/img/rename-game-exe.png)
+#### Change the title of the game window:**
```
change to 
`####  Change the title of the game window`

Snailedlt commented on docs/create-handlers.md:
```diff
@@ -2,15 +2,330 @@
+### Start multiple instances
+First thing, you must find a way to make the game run multiple times.
+If it doesn't run twice, these are the methods that I know (*For some games, you will need to combine these methods):
+#### Closing game mutexes:
```
remove colon at the end

Snailedlt commented on docs/create-handlers.md:
```diff
@@ -2,15 +2,330 @@
+If you think to have found the right one, right-click on it and choose ***"Close Handle"***. After that, try to run a second instance of the game and see if it opens. (some games require multiple mutexes to be closed).
+Be careful that launchers check if the game is already running. For example, Goldberg, make it unseen from Steam.
+#### Check public scripts for examples.
+#### Rename the game exe:
```
remove ending colon

Snailedlt commented on docs/create-handlers.md:
```diff
@@ -2,15 +2,330 @@
+#### Change the title of the game window:**
+![Change Title of the game window](../static/img/change-title.jpg)
+You can use https://www.nirsoft.net/utils/winexp.html for testing. (in Nucleus-Coop = Game.IdInWindowTitle = true;) 
+#### Start the game as a different windows user:**
```
change to 
`#### Start the game as a different windows user`

Snailedlt commented on docs/create-handlers.md:
```diff
@@ -2,15 +2,330 @@
+You can find most of the lines in the Mod-ReadMe.txt (inside Nucleus-Coop root folder) and in the Master_SCript.js (documentation link inside Reddit FAQ post).
+Also, all my public scripts (Bizzo) have descriptions inside that could help you.
+
+Here some tips and essential lines:
```
remove this line

Snailedlt commented on docs/create-handlers.md:
```diff
@@ -2,15 +2,330 @@
+Game.MaxPlayersOneMonitor = 4; |	This is just an info. It will not limit the players' numbers.
+Game.MaxPlayers = 16; | 	This is just an info. Usually, we write the max players number the game support. 
+
+#### Other important ones:
```
remove ending colon `:`

Snailedlt commented on docs/create-handlers.md:
```diff
@@ -2,15 +2,330 @@
+
+### Handlers
+
+#### This is needed:
```
remove ending colon

Snailedlt commented on docs/create-handlers.md:
```diff
@@ -2,15 +2,330 @@
+---- | ----------
+Game.HandlerInterval = 100; | The interval in milliseconds the Handler should be updated at. Set it to 0 to disable updating (will lose all functionality that depends on ticks).
+
+#### Higher time could help low spec PCs:
```
ending colon

Snailedlt commented on docs/create-handlers.md:
```diff
@@ -2,15 +2,330 @@
+Inside "Game.Play = function() {" you can write operations that Nucleus will do in the context of the game starting. The lines inside this section are unique on each script.
+I invite you to check the documentation inside the FAQ Reddit post or check other scripts to see examples. As I said, mine (Bizzo) as descriptions inside that could help you.
+
+### Here some examples:
```
remove `:`

Snailedlt reviewed (commented):
Alright I think these are the last changes you have to make in order for it to be as close to the original guide as possible. Then after that's sorted out, we can start improving it a bit 👍 

Well done so far 🥇 

Let me know if anything is unclear, or if you disagree with any of the changes I suggested :)

Lord-Baldwin-of-Kerry commented:
@Snailedlt I implemented all the observations you made, and I managed to catch a few discrepancies between my commit and the original guide. I fixed them :D

Snailedlt commented:
Good!
Only thing I can see now is this:
![image](https://user-images.githubusercontent.com/43886029/129281795-121d51d7-cde1-4e58-8beb-067077f01a37.png)

I'll merge the request, and fix it in my own afterwards :)
Well done

Snailedlt commented:
@Lord-Baldwin-of-Kerry published now, the topmost imagelink doesn't work on the server. Which is odd, beucase it works just fine locally.

Lord-Baldwin-of-Kerry commented:
> @Lord-Baldwin-of-Kerry published now, the topmost imagelink doesn't work on the server. Which is odd, beucase it works just fine locally.

Interesting, that is the only image that uses a img html tag, the other ones are using markdown. My first guest is that the path gets changed in the compiling process, I can do a quick pull request changing the img src to a relative path, hopefully that should fix it! :D



> Good!
> Only thing I can see now is this:
> ![image](https://user-images.githubusercontent.com/43886029/129281795-121d51d7-cde1-4e58-8beb-067077f01a37.png)
> 
> I'll merge the request, and fix it in my own afterwards :)
> Well done

for that I guess we could use the <br /> tags after each image
